datalake/src/main/java/com/sequenceiq/datalake/flow/detach/SdxDetachActions.java

```diff
@@ -1,173 +1,291 @@
+import com.sequenceiq.datalake.service.sdx.attach.SdxAttachService;
+import com.sequenceiq.datalake.service.sdx.attach.SdxDetachService;
 import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
-import com.sequenceiq.flow.core.Flow;
 import com.sequenceiq.flow.core.FlowEvent;
 import com.sequenceiq.flow.core.FlowParameters;
 import com.sequenceiq.flow.core.FlowState;
-import com.sequenceiq.redbeams.api.endpoint.v4.databaseserver.DatabaseServerV4Endpoint;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.StateContext;
+import org.springframework.statemachine.action.Action;
 
 @Configuration
 public class SdxDetachActions {
-
     private static final Logger LOGGER = LoggerFactory.getLogger(SdxDetachActions.class);
 
     private static final String RESIZED_SDX = "RESIZED_SDX";
 
-    private static final String DETACHING_SDX = "DETACHING_SDX";
-
-    private static final String SDX_NAME = "SDX_NAME";
+    private static final String DETACHED_SDX = "DETACHED_SDX";
 
     @Inject
     private SdxStatusService sdxStatusService;
 
     @Inject
-    private SdxDetachService sdxDetachService;
+    private SdxClusterRepository sdxClusterRepository;
 
     @Inject
-    private OwnerAssignmentService ownerAssignmentService;
+    private SdxDetachService sdxDetachService;
 
     @Inject
-    private TransactionService transactionService;
+    private SdxAttachService sdxAttachService;
 
-    @Inject
-    private SdxClusterRepository sdxClusterRepository;
+    @Bean(name = "SDX_DETACH_CLUSTER_STATE")
+    public Action<?, ?> sdxDetachCluster() {
+        return new AbstractSdxAction<>(SdxStartDetachEvent.class) {
+            @Override
+            protected SdxContext createFlowContext(FlowParameters flowParameters,
+                    StateContext<FlowState, FlowEvent> stateContext,
+                    SdxStartDetachEvent payload) {
+                return SdxContext.from(flowParameters, payload);
+            }
 
-    @Inject
-    private StackV4Endpoint stackV4Endpoint;
+            @Override
+            protected void doExecute(SdxContext context, SdxStartDetachEvent payload,
+                    Map<Object, Object> variables) throws Exception {
+                LOGGER.info("Detaching of SDX with ID {} in progress.", payload.getResourceId());
+                sdxStatusService.setStatusForDatalakeAndNotify(
+                        DatalakeStatusEnum.STOPPED, "Datalake detach in progress.", payload.getResourceId()
+                );
+
+                SdxCluster cluster = sdxClusterRepository.findById(payload.getResourceId())
+                        .orElseThrow(() -> new NotFoundException("SdxCluster was not found for ID: " + payload.getResourceId()));
```

Comment:
these can be moved into SdxDetachService I think

Reply:
Done.

Reply:
thanks!